src/redux/actions/bookmarkActions.js

```diff
@@ -0,0 +1,26 @@
+import { toast } from 'react-toastify';
+import { ADD_BOOKMARK, REMOVE_BOOKMARK } from './actionTypes';
+import { sendHttpRequest } from '../../utils/index';
+
+const addBookmark = id => async (dispatch) => {
+  try {
+    const response = await sendHttpRequest(`articles/${id}/bookmark`, 'post');
+    if (response.bookmark) {
+      toast.success(response.message);
+      return dispatch({ type: ADD_BOOKMARK, payload: response.bookmark });
+    }
+    toast.success(response.message);
+    return dispatch({ type: REMOVE_BOOKMARK, id });
+  } catch (error) {
+    switch (error.response.status) {
+      case 404:
+        return toast.error(error.response.data.message);
+      case 401:
+        return toast.error(error.response.data.message);
```

Comment:
Keep things DRY
```suggestion
      case 404:
      case 401:
        return toast.error(error.response.data.message);
```

Reply:
You can cater for both 404 and 401 with one return.

Reply:
Okay